engines/asylum/console.cpp

```diff
@@ -0,0 +1,985 @@
+			debugPrintf("Syntax: %s <type> (nothing: all  -  1: show set flags  -  0: show unset flags)\n", argv[0]);
+			return true;
+		}
+
+		// Show only set/unset flags
+		int count = 0;
+		for (int32 i = 0; i < 1512; i++) {
+			if (_vm->isGameFlagSet((GameFlag)i) == (bool)type) {
+				debugPrintf("%04d: %d    ", i, _vm->isGameFlagSet((GameFlag)i));
+				++count;
+			}
+
+			if ((count + 1) % 10 == 0)
+				debugPrintf("\n");
+		}
+		debugPrintf("\n\n%s flags: %d\n", (type ? "Set" : "Unset"), count);
+	}
+
+	return true;
+}
+
+bool Console::cmdShowWorldStats(int32, const char **) {
+	debugPrintf("WorldStats\n");
+	debugPrintf("----------\n");
+	debugPrintf("%s", getWorld()->toString().c_str());
+
+	return true;
+}
+
+bool Console::cmdShowObject(int32 argc, const char **argv) {
+	if (argc != 3) {
+		debugPrintf("Syntax: %s [id|idx] <target>\n", argv[0]);
+		return true;
+	}
+
+	if (Common::String(argv[1]) == "id") {
+		int id = atoi(argv[2]);
+		for (uint32 i = 0; i < getWorld()->objects.size(); i++) {
+			if (getWorld()->objects[i]->getId() == id) {
+				debugPrintf("%s", getWorld()->objects[i]->toString(false).c_str());
+				return true;
+			}
+		}
+		debugPrintf("No object with id %d found\n", id);
+	} else if (Common::String(argv[1]) == "idx") {
+		int index = atoi(argv[2]);
+		int maxIndex = getWorld()->objects.size() - 1;
+
+		if (index < 0 || index > maxIndex) {
+			debugPrintf("[error] index should be between 0 and %d\n", maxIndex);
+			return true;
+		}
+
+		debugPrintf("%s", getWorld()->objects[index]->toString(false).c_str());
+
+	} else {
+		debugPrintf("[error] valid options are 'id' and 'idx'\n");
+	}
+
+	return true;
+}
+
+bool Console::cmdShowAction(int32 argc, const char **argv) {
+	if (argc != 3) {
+		debugPrintf("Syntax: %s [id|idx] <target>\n", argv[0]);
+		return true;
+	}
+
+	if (Common::String(argv[1]) == "id") {
+		int id = atoi(argv[2]);
+		for (uint32 i = 0; i < getWorld()->actions.size(); i++) {
+			if (getWorld()->actions[i]->id == id) {
+				debugPrintf("%s", getWorld()->actions[i]->toString().c_str());
+				return true;
+			}
+		}
+		debugPrintf("No action with id %d found\n", id);
+	} else if (Common::String(argv[1]) == "idx") {
+		int index = atoi(argv[2]);
+		int maxIndex = getWorld()->actions.size() - 1;
+
+		if (index < 0 || index > maxIndex) {
+			debugPrintf("[error] index should be between 0 and %d\n", maxIndex);
+			return true;
+		}
+
+		debugPrintf("%s", getWorld()->actions[index]->toString().c_str());
+
+	} else {
+		debugPrintf("[error] valid options are 'id' and 'idx'\n");
+	}
+
+	return true;
+}
+
+bool Console::cmdListObjects(int32 argc, const char **argv) {
+	if (argc != 2) {
+		debugPrintf("Syntax: %s [onscreen|*]\n", argv[0]);
+		return true;
+	}
+
+	if (argc == 2) {
+		if (Common::String(argv[1]) == "onscreen") {
+			for (uint32 i = 0; i < getWorld()->objects.size(); i++) {
+				if (getWorld()->objects[i]->isOnScreen()) {
+					debugPrintf("%s", getWorld()->objects[i]->toString().c_str());
```

Comment:
The formatting might be off here - looks like there is either a newline or a space missing

Reply:
Object::toString() appends a newline to the resulting string